inngest/src/main/kotlin/com/inngest/Comm.kt

```diff
@@ -166,8 +169,50 @@ class CommHandler(
     // TODO
 //    fun sync(): Result<InngestSyncResult> = Result.success(InngestSyncResult.None)
 
-    fun introspect(origin: String): String {
-        val requestPayload = getRegistrationRequestPayload(origin)
+    fun introspect(
+        signature: String?,
+        requestBody: String,
+        serverKind: String?,
+    ): String {
+        val insecureIntrospection =
+            InsecureIntrospection(
+                functionCount = functions.size,
+                hasEventKey = Environment.isInngestEventKeySet(client.eventKey),
+                hasSigningKey = config.hasSigningKey(),
+                mode = if (client.env == InngestEnv.Dev) "dev" else "cloud",
+            )
+
+        val requestPayload =
+            when (client.env) {
+                InngestEnv.Dev -> insecureIntrospection
+
+                else ->
+                    runCatching {
```

Comment:
til, neat combination with the getOrElse

Reply:
yeah it fits perfect with this example of only one function that could throw.